pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/Patch.java

```diff
@@ -161,34 +180,66 @@ public static boolean patch(@NonNull String id, @NonNull String user, @NonNull N
-                try {
-                    ResultSet rs = stmt.executeQuery();
-                    if (rs.next()) {
-                        dbVals.resultValue = false;
-                    } else {
-                        dbVals.resultValue = true;
-                    }
-                } catch (Exception e) {
-                    dbVals.resultException = e;
-                }
-            } catch (SQLException ex) {
-                dbVals.resultException = ex;
-            }
-            return dbVals;
-        });
-        return (Boolean) results.resultValue;
+    boolean isOncHistoryDeletePatch() {
+        return getNameValue().getName().equals(ONC_HISTORY_DELETED_FIELD) &&
+                getTableAlias().equals(DBConstants.DDP_ONC_HISTORY_DETAIL_ALIAS);
+    }
+
+     boolean isTissueDeletePatch() {
+        return getNameValue().getName().equals(TISSUE_DELETED_FIELD) && getTableAlias().equals(DBConstants.DDP_TISSUE_ALIAS);
+    }
+
+    //creates a list of patches for deleting tissues that belong to an onc history
+    protected static List<Patch> getPatchForTissues(Patch oncHistoryPatch) {
+        List<Integer> tissueIds = TissueDao.getTissuesByOncHistoryDetailId(oncHistoryPatch.getIdAsInt());
+        List<Patch> deletePatches = new ArrayList<>();
+        for (Integer tissueId : tissueIds) {
+            NameValue nameValue = new NameValue(TISSUE_DELETED_FIELD, TRUE_FLAG);
+            Patch patch =
+                    new Patch(String.valueOf(tissueId), OncHistoryDetail.ONC_HISTORY_DETAIL_ID, oncHistoryPatch.getId(),
+                            oncHistoryPatch.getUser(), nameValue, null, true,
+                            oncHistoryPatch.getDdpParticipantId(), oncHistoryPatch.getRealm(), DBConstants.DDP_TISSUE_ALIAS);
+            patch.setTableAlias(DBConstants.DDP_TISSUE_ALIAS);
+            deletePatches.add(patch);
+        }
+        return deletePatches;
+    }
+
+    //creates a list of patches for deleting the sm ids that belong to a tissue
+    protected static List<Patch> getPatchForSmIds(Patch tissuePatch) {
+        List<String> smIds = TissueSMIDDao.getSmIdPksForTissue(tissuePatch.getId());
+        List<Patch> deletePatches = new ArrayList<>();
+        for (String smIdPk : smIds) {
+            NameValue nameValue = new NameValue(SM_ID_DELETED_FIELD, TRUE_FLAG);
+            deletePatches.add(
+                    new Patch(smIdPk, TissuePatch.TISSUE_ID, tissuePatch.getId(), tissuePatch.getUser(), nameValue, null, true,
+                            tissuePatch.getDdpParticipantId(), tissuePatch.getRealm(), DBConstants.SM_ID_TABLE_ALIAS));
+        }
+        return deletePatches;
+    }
+
+    public boolean hasDDPParticipantId() {
+        return StringUtils.isNotBlank(this.ddpParticipantId);
+    }
+
+    public boolean isSmIdDeletePatch() {
+        return (this.getNameValue().getName().contains(".deleted")) &&
```

Comment:
Since this is now an instance method, you can skip the getter and just use `nameValue` instead of `getNameValue()`.